paymentsheet/src/test/java/com/stripe/android/paymentsheet/SavedPaymentMethodsExtensionTest.kt

```diff
@@ -0,0 +1,100 @@
+package com.stripe.android.paymentsheet
+
+import android.content.Context
+import com.stripe.android.core.strings.ResolvableString
+import com.stripe.android.model.PaymentMethodCode
+import com.stripe.android.paymentsheet.verticalmode.toDisplayableSavedPaymentMethod
+import com.stripe.android.testing.PaymentMethodFactory
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.`when`
+import kotlin.test.assertEquals
+
+class SavedPaymentMethodsExtensionTest {
+
+    @Test
+    fun `DisplayableSavedPaymentMethod is created correctly when metadata & defaultPaymentMethodId is null`() {
+        val context = mock<Context>()
+        val resolvableStringValue = "abcde123"
+        val resolvableString = mock<ResolvableString>()
+        `when`(resolvableString.resolve(context)).thenReturn(resolvableStringValue)
+        val providePaymentMethodName: (PaymentMethodCode?) -> ResolvableString = { pmc ->
+            resolvableString
+        }
+
+        val paymentMethodId = "aaa111"
+        val paymentMethod = PaymentMethodFactory.card(paymentMethodId)
+
+        assertEquals(
```

Comment:
for the tests in this file, can you assert on just the individual fields we need to test, instead of on the whole object? ie I think in these tests, we really only need to test that shouldShowDefaultBadge is what we expect it to be, not the whole object